src/nupic/engine/Output.cpp

```diff
@@ -91,20 +92,17 @@ Dimensions Output::determineDimensions() {
         dim_.push_back(0);  // set Don't care.
       }
     }
-
-    // If we still have a isDontcare, check if the spec defines 
-    // an input on this region to get the dimensions from. 
-    if (dim_.isDontcare()) {
-      std::string inheritFrom = srcSpec->outputs.getByName(name_).inheritFrom;
-      if (!inheritFrom.empty()) {
-        Input* in = region_->getInput(inheritFrom);
-        if (in) {
-          Dimensions d = in->getDimensions();
-          if (d.isSpecified()) {
-            dim_ = d;
-          }
-        }
-      }
+  }
+  // If we still have a isDontcare, check if the spec defines 
+  // regionLevel then get the dimensions from the region dims. 
+  bool regionLevel = srcSpec->outputs.getByName(name_).regionLevel;
+  if (regionLevel) {
+    Dimensions d = region_->getDimensions();
+    if (dim_.isDontcare()&& d.isSpecified()) {
+      dim_ = d;
+    }
+    else if (dim_.isSpecified() && d.isDontcare()) {
+      region_->setDimensions(dim_);
```

Comment:
this is some mistake, as those 2 are the same conditions.

Reply:
still this one to address. see the if-elseif condition

Reply:
No, not the same.
```
    if (dim_.isDontcare()&& d.isSpecified()) {
      dim_ = d;
    }
    else if (dim_.isSpecified() && d.isDontcare()) {
      region_->setDimensions(dim_);
    }
```

Reply:
I'm sorry, that got me confused!